src/gasman.c

```diff
@@ -1883,88 +1875,6 @@ UInt CollectBags (
 
     /* * * * * * * * * * * * * * * sweep phase * * * * * * * * * * * * * * */
 
-#if 0
-    /* call freeing function for all dead bags                             */
-    if ( NrTabFreeFuncBags ) {
```

Comment:
As a rule, it would be nice if we had a finalizer facility that wasn't limited to just the Boehm GC. Finalization is generally useful/necessary for packages that wrap external resources and for interaction with other programming languages, even though GAP proper does not have a direct need for them.